lib/crowbar/client/command/proposal/edit.rb

```diff
@@ -140,6 +150,29 @@ def payload_content
               from_editor
             end
           end
+
+          def check_nodes(configuration)
+            avaliable_nodes = JSON.parse(Request::Node::List.new({}).process)
+            if avaliable_nodes["nodes"].nil?
+              say "Internal error: Invalid node list provided"
+              raise
+            end
+
+            avaliable_nodes = avaliable_nodes["nodes"]
+
+            configuration["deployment"].each do |service_name, service|
+              next if service["elements"].nil?
+
+              service["elements"].each do |role, nodes|
+                nodes.each do |node|
+                  if avaliable_nodes.select { |item| item["name"] == node }.empty?
+                    say "ERROR : Node #{node} is not available in the list of possible nodes, "\
```

Comment:
IMO it would be nicer to collect the names of bad nodes and print one message at the end with full list of nodes.

Also consider not sending the request at all if you detect invalid nodes. This way user will not get the 500 result from the server.

Reply:
Thanks @skazi0. They was useful comments